File: Learn Objective-C for Java Developers/Chapter 13/ObjC Greeter/Greeter/Greeter.m
//
//  Greeter.m
//  Greeter
//
//  Created by James Bucanek on 4/7/09.
//  Copyright 2009 Apress. All rights reserved.
//

#import "Greeter.h"

#import "Guest.h"

@implementation Greeter

- (void)sayHello
{
	NSLog(@"Greeter %@ was asked to sayHello",self);
}

- (void)greetGuest:(bycopy Guest*)guest
{
	NSLog(@"Greeter %@ was asked to greetGuest:%@",self,guest);
	[guest listen:[NSString stringWithFormat:@"I'm pleased to meet you, %@!",guest]];
}

- (NSString*)sayGoodbye
{
	NSLog(@"Greeter %@ was asked to sayGoodbye",self);
	return @"It was a pleasure serving you.";
}

@end

File: Learn Objective-C for Java Developers/Chapter 13/Java Greeter/com/apress/java/rmi/Greeter.java
package com.apress.java.rmi;

import java.rmi.*;

public interface Greeter extends Remote {
	public void sayHello( ) throws java.rmi.RemoteException;
	public void greetGuest( Guest listener ) throws java.rmi.RemoteException;
	public String sayGoodbye( ) throws java.rmi.RemoteException;
}
